### ISSUE TITLE
[sw,plic irq test] Enable the autogen'ed PLIC IRQ test on Verilator and FPGA

### ISSUE THREAD

**Issue by sriyerg** [lowRISC/opentitan#8656]
This test is auto-generated by `topgen`. It resides here: `sw/device/tests/autogen/plic_all_irqs_test.c`. It is already enabled and passing in DV. Since it touches (almost) all blocks, it will serve as a pretty good smoke test to run in CI.

I tried running it locally but it unfortunately fails on Verilator. 
```
IE00000 plic_all_irqs_test.c:154] CHECK-fail: Interrupt from incorrect peripheral: exp = 16, obs = 1
I00001 test_status.c:34] FAIL!
```

This basically means on the the Verilator build, some IP is generating a spurious interrupt that the test does not expect. 

Hoping someone with spare cycles can debug this and get it working on Verilator. 

```bash
$ cd $REPO_TOP
$ fusesoc --cores-root . run --flag=fileset_top --target=sim --setup --build lowrisc:dv:chip_verilator_sim
$ ./meson_init.sh
$ ninja -C build-out/ all
$ ./build/lowrisc_dv_chip_verilator_sim_0.1/sim-verilator/Vchip_sim_tb -meminit=rom,build-bin/sw/device/boot_rom/boot_rom_sim_verilator.scr.39.vmem --meminit=otp,build-bin/sw/device/otp_img/otp_img_sim_verilator.vmem --meminit=flash,build-bin/sw/device/tests/plic_all_irqs_test_sim_verilator
```

**Comment by sriyerg** [comment]
The DV setup circumvents the use of on-device resources for logging, which the Verilator and FPGA setups cannot. The latter use UART0 for logging. Consequently, if the PLIC all IRQs test enables UART0 interrupts, the logging mechanism ends up triggering an interrupt (TX empty, for example, when UART0 is done transferring all characters in a log) inadvertently, when the test is expecting an interrupt from some other peripheral. 

The solution here is to enable UART0 to test the interrupts only for DV and skip for the other platforms. This test is now passing. 
Fixed in #10183.

**Comment by tjaychen** [comment]
sounds great!
